Sources/ArgumentParser/Parsing/Interactive.swift

```diff
@@ -57,51 +57,91 @@ extension CommandParser {
+                                              updateBy: (String) throws -> Void,
+                                              arguments: ArgumentSet)
+  {
     print("? Please enter '\(label)': ", terminator: "")
-    return getInput()?.components(separatedBy: " ") ?? [""]
+    let strs = getInput()?.components(separatedBy: " ") ?? [""]
+
+    for str in strs {
+      do {
+        try updateBy(str)
+      } catch {
+        // Handle ParserError
+        guard let error = error as? ParserError else { return }
+        let generator = ErrorMessageGenerator(arguments: arguments, error: error)
+        let description = generator.makeErrorMessage() ?? error.localizedDescription
+        guard case let .unableToParseValue(_, _, original, _, _) = error else { return }
+        print("Error: " + description + ".\n")
+        replaceInvalidValue(original: original, updateBy: updateBy, arguments: arguments)
+      }
+    }
   }
-  
-  fileprivate mutating func getCaseIterableEnum(label: String, allValues: [String]) -> [String] {
+
+  fileprivate mutating func replaceInvalidValue(original: String,
+                                                updateBy: (String) throws -> Void,
+                                                arguments: ArgumentSet)
+  {
+    print("? Please replace '\(original)': ", terminator: "")
+    let input = getInput() ?? ""
+
+    do {
+      try updateBy(input)
+    } catch {
+      // Handle ParserError
+      guard let error = error as? ParserError else { return }
+      let generator = ErrorMessageGenerator(arguments: arguments, error: error)
+      let description = generator.makeErrorMessage() ?? error.localizedDescription
+      guard case let .unableToParseValue(_, _, original, _, _) = error else { return }
+      print("Error: " + description + ".\n")
+      replaceInvalidValue(original: original, updateBy: updateBy, arguments: arguments)
+    }
+  }
+
+  fileprivate mutating func storeCaseIterableEnums(label: String,
+                                                   allValues: [String],
+                                                   updateBy: (String) throws -> Void,
+                                                   arguments: ArgumentSet)
+  {
     print("? Please select '\(label)': ", terminator: "")
     let strs = getInput()?.components(separatedBy: " ") ?? [""]
-    
+
     var nums: [String] = []
     let range = 1 ... allValues.count
-    for str in strs {
-      guard let index = Int(str) else {
-        print("Error: '\(str)' is not a serial number.\n")
-        return getCaseIterableEnum(label: label, allValues: allValues)
+    for num in strs {
+      guard let index = Int(num) else {
+        print("Error: '\(num)' is not a serial number.\n")
+        storeCaseIterableEnums(label: label, allValues: allValues, updateBy: updateBy, arguments: arguments)
+        return
       }
 
       guard range.contains(index) else {
         print("Error: '\(index)' is not in the range of \(range) \n")
```

Comment:
We want to avoid showing programming concepts like `3...6` in error messages. Maybe something like this, or maybe two different messages for too low and too high? For example, `'x' must be at least '1'` and `'x' must be at most '7'`. What do you think?
```suggestion
        print("Error: '\(index)' is not in the range '\(range.lowerBound) - \(range.upperBound)'.\n")
```

Reply:
"is not in the range 3 - 6." provides more complete information.